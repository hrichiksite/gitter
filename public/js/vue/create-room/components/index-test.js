jest.mock('../../../utils/appevents');

const mount = require('../../__test__/vuex-mount');
const { default: Index } = require('./index.vue');
const appEvents = require('../../../utils/appevents');

describe('CreateRoom', () => {
  beforeEach(() => {
    appEvents.trigger.mockReset();
  });

  it('matches snapshot', () => {
    const { wrapper } = mount(Index);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room submit loading matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      store.state.createRoom.roomSubmitRequest = { loading: true };
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room submit error matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      store.state.createRoom.roomSubmitRequest = { error: 'some-error' };
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('clicking submit fires store action', () => {
    const { wrapper, stubbedActions } = mount(Index);

    wrapper.find({ ref: 'submitButton' }).trigger('click');

    expect(stubbedActions.createRoom.submitRoom).toHaveBeenCalledWith(
      expect.anything(),
      expect.anything(),
      undefined
    );
  });

  it('clicking on the modal does NOT close the modal', () => {
    const { wrapper } = mount(Index);

    wrapper.find({ ref: 'modal' }).trigger('click');

    expect(appEvents.trigger.mock.calls[0]).toEqual(undefined);
  });

  it('closes modal when close button is clicked', () => {
    const { wrapper } = mount(Index);

    wrapper.find({ ref: 'closeButton' }).trigger('click');

    expect(appEvents.trigger.mock.calls[0]).toEqual(['destroy-create-room-view']);
  });

  it('closes modal when modal backdrop is clicked', () => {
    const { wrapper } = mount(Index);

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(appEvents.trigger.mock.calls[0]).toEqual(['destroy-create-room-view']);
  });
});
