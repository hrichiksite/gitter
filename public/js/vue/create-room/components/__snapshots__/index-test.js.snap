// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoom matches snapshot 1`] = `
<div
  class="root modal-backdrop js-create-room-view-root"
>
  <div
    class="modal--default"
  >
    <header
      class="modal--default__header"
    >
      <h1
        class="modal--default__header__title js-modal-title-text"
      >
        Create a room
      </h1>
       
      <span
        class="modal--default__header__icon icon-cancel-circle"
        data-action="close"
      />
    </header>
     
    <section
      class="modal--default__content"
    >
      <p
        class="create-room-intro-text"
      >
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/rooms.md"
          target="_blank"
        >
          Gitter rooms
        </a>
        
        exist inside of
        
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/communities.md"
          target="_blank"
        >
          communities
        </a>
        . For example, if you want to end up with the URI:
        
        <strong>
          https://gitter.im/badger/webapp
        </strong>
        , first create a
        
        <strong>
          badger
        </strong>
         community, then create a 
        <strong>
          webapp
        </strong>
         room under that
        community. You can create many rooms under your community.
      
      </p>
       
      <p
        class="create-room-intro-text"
      >
        
        If you need to create a community to put your rooms in first, jump to the
        
        <a
          href="#createcommunity"
        >
          community create flow
        </a>
      </p>
       
      <hr />
       
      <group-and-room-input-section-stub />
       
      <repo-select-section-stub />
       
      <hr />
       
      <security-select-section-stub />
       
      <create-room-extra-options-stub />
       
      <!---->
    </section>
     
    <footer
      class="modal--default__footer-wrapper"
    >
      <div
        class="modal--default__footer"
      >
        <button
          class="modal--default__footer__btn pull-right"
        >
          <!---->
           Create
        
        </button>
      </div>
    </footer>
  </div>
</div>
`;

exports[`CreateRoom room submit error matches snapshot 1`] = `
<div
  class="root modal-backdrop js-create-room-view-root"
>
  <div
    class="modal--default"
  >
    <header
      class="modal--default__header"
    >
      <h1
        class="modal--default__header__title js-modal-title-text"
      >
        Create a room
      </h1>
       
      <span
        class="modal--default__header__icon icon-cancel-circle"
        data-action="close"
      />
    </header>
     
    <section
      class="modal--default__content"
    >
      <p
        class="create-room-intro-text"
      >
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/rooms.md"
          target="_blank"
        >
          Gitter rooms
        </a>
        
        exist inside of
        
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/communities.md"
          target="_blank"
        >
          communities
        </a>
        . For example, if you want to end up with the URI:
        
        <strong>
          https://gitter.im/badger/webapp
        </strong>
        , first create a
        
        <strong>
          badger
        </strong>
         community, then create a 
        <strong>
          webapp
        </strong>
         room under that
        community. You can create many rooms under your community.
      
      </p>
       
      <p
        class="create-room-intro-text"
      >
        
        If you need to create a community to put your rooms in first, jump to the
        
        <a
          href="#createcommunity"
        >
          community create flow
        </a>
      </p>
       
      <hr />
       
      <group-and-room-input-section-stub />
       
      <repo-select-section-stub />
       
      <hr />
       
      <security-select-section-stub />
       
      <create-room-extra-options-stub />
       
      <section
        class="error-section"
      >
        
        some-error
      
      </section>
    </section>
     
    <footer
      class="modal--default__footer-wrapper"
    >
      <div
        class="modal--default__footer"
      >
        <button
          class="modal--default__footer__btn pull-right"
        >
          <!---->
           Create
        
        </button>
      </div>
    </footer>
  </div>
</div>
`;

exports[`CreateRoom room submit loading matches snapshot 1`] = `
<div
  class="root modal-backdrop js-create-room-view-root"
>
  <div
    class="modal--default"
  >
    <header
      class="modal--default__header"
    >
      <h1
        class="modal--default__header__title js-modal-title-text"
      >
        Create a room
      </h1>
       
      <span
        class="modal--default__header__icon icon-cancel-circle"
        data-action="close"
      />
    </header>
     
    <section
      class="modal--default__content"
    >
      <p
        class="create-room-intro-text"
      >
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/rooms.md"
          target="_blank"
        >
          Gitter rooms
        </a>
        
        exist inside of
        
        <a
          href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/communities.md"
          target="_blank"
        >
          communities
        </a>
        . For example, if you want to end up with the URI:
        
        <strong>
          https://gitter.im/badger/webapp
        </strong>
        , first create a
        
        <strong>
          badger
        </strong>
         community, then create a 
        <strong>
          webapp
        </strong>
         room under that
        community. You can create many rooms under your community.
      
      </p>
       
      <p
        class="create-room-intro-text"
      >
        
        If you need to create a community to put your rooms in first, jump to the
        
        <a
          href="#createcommunity"
        >
          community create flow
        </a>
      </p>
       
      <hr />
       
      <group-and-room-input-section-stub />
       
      <repo-select-section-stub />
       
      <hr />
       
      <security-select-section-stub />
       
      <create-room-extra-options-stub />
       
      <!---->
    </section>
     
    <footer
      class="modal--default__footer-wrapper"
    >
      <div
        class="modal--default__footer"
      >
        <button
          class="modal--default__footer__btn pull-right"
          disabled="disabled"
        >
          <loading-spinner-stub />
           Create
        
        </button>
      </div>
    </footer>
  </div>
</div>
`;
